carts/views.py

```diff
@@ -1,3 +1,87 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http.response import JsonResponse
+from django.views         import View
+from django.db.models import Q
+
+from .models          import Cart
+from products.models  import Product
+from users.models     import User
+from users.decorators import login_decorator
+
+class CartView(View):
+    @login_decorator
+    def post(self, request):
+        try:
+            data       = json.loads(request.body)
+            user       = request.user
+            product_id = data['product_id']
+            quantity   = data['quantity']
+
+            if not Product.objects.filter(id=product_id).exists():
+                return JsonResponse({'MESSAGE':'DOES_NOT_EXIST_PRODUCT'}, status = 404)
+
+            Cart.objects.create(
+                user       = user,
+                product_id = product_id,
+                quantity   = quantity
+            )
+            return JsonResponse({'MESSAGE':'CREATE'}, status = 201)
+
+        except KeyError:
+            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status = 400)
+        except ValueError:
+            return JsonResponse({'MESSAGE':'VALUE_ERROR'}, status = 400)
+
+    @login_decorator
+    def get(self, request):
+
+        user = request.user
+        carts = Cart.objects.filter(user_id=user).select_related('product')
+        
+        result = [{
+            'name' : cart.product.name,
+            'price': int(cart.product.price),
+            'image': [image.url for image in cart.product.image_set.all()],
+            'quantity' : cart.quantity
+        }for cart in carts]
+
+        return JsonResponse({'result':result}, status = 201)
+
+    @login_decorator
+    def delete(self, request):
+        try:
+            data = json.loads(request.body)
+
+            products = data['product_id']
+
+            q = Q()
+
+            for request_product_id in products:
+                q |= Q(product_id = request_product_id)
+            q &= Q(user_id = request.user.id)
+            
+            Cart.objects.filter(q).delete()
+
+            return JsonResponse({'MESSAGE':'SUCCESS'}, status=201)
+
+        except KeyError:
+            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
```

Comment:
DELETE의 경우 request message body를 사용할 수 없습니다!

1. REST API에서는 DELETE로 하나의 데이터를 삭제할 경우 path parameter를 사용해서 구현하셔야합니다.
`ex) DELETE /cart/1`

2. 여러개의 데이터를 한번에 삭제할 경우 query parameter를 사용해서 구현하는 방법을 많이 사용하고 있습니다.
`ex) DELETE /cart?product_ids = [1,2,3]
 
3. 자원 삭제시 사용되는 http status code가 무엇인지 찾아보고 정확하게 사용해주세요~
 
```suggestion
    @login_decorator
    def delete(self, request):
        try:            
            Cart.objects.filter(user_id = .., product_id__in ...).delete()

            return JsonResponse({'MESSAGE':'NO_CONTENT'}, status=204)

        except KeyError:
            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
```

Reply:
넵. 말씀해주신대로 고치겠습니다.
PS. 크롬 로컬에서 React 돌리고 DB까지 테스트 한 뒤 데이터 확인하고 올렸습니다.

Reply:
병민님, 논의된대로 1개 삭제와 리스트 삭제를 통합해서 사용하겠습니다!
.../?product_id=[1]
.../?product_id=[1,2,3]
처럼 url query parameter에 포함될거에요!